src/shared/codelens/goCodeLensProvider.ts

```diff
@@ -0,0 +1,179 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { VSCODE_EXTENSION_ID } from '../extensions'
+import { LambdaHandlerCandidate } from '../lambdaHandlerSearch'
+import { getLogger } from '../logger/logger'
+import { stripNewLinesAndComments } from '../../shared/utilities/textUtilities'
+import { findParentProjectFile } from '../utilities/workspaceUtils'
+
+export const GO_LANGUAGE = 'go'
+export const GO_ALLFILES: vscode.DocumentFilter[] = [
+    {
+        scheme: 'file',
+        language: GO_LANGUAGE,
+    },
+]
+
+// Go modules were introduced in Go 1.11
+// Before that, $GOPATH had to be set properly to find dependencies
+// We currently just ignore projects without a Go module file
+export const GO_BASE_PATTERN = '**/go.mod'
```

Comment:
SAM cli requires a go.mod file so we don't need to worry about support before then for run+debug